upload-api/stores/agent.js

```diff
@@ -0,0 +1,118 @@
+import * as API from '../types.js'
+import * as Store from './agent/store.js'
+import * as Stream from './agent/stream.js'
+
+export { API }
+
+/**
+ * @typedef {object} Options
+ * @property {Stream.Options} stream
+ * @property {Store.Options} store
+ *
+ * @typedef {object} Connection
+ * @property {Store.Store} store
+ * @property {Stream.Stream} stream
+ *
+ *
+ * @param {Options} options
+ */
+export const open = (options) =>
+  new AgentMessageStore({
+    store: Store.open(options.store),
+    stream: Stream.open(options.stream),
+  })
+
+/**
+ * @implements {API.AgentStore}
+ */
+class AgentMessageStore {
+  /**
+   * @param {Connection} connection
+   */
+  constructor(connection) {
+    this.connection = connection
+    this.invocations = new InvocationsIndex(connection.store)
+    this.receipts = new ReceiptsIndex(connection.store)
+  }
+
+  get messages() {
+    return this
+  }
+
+  /**
+   * @param {API.ParsedAgentMessage} message
+   * @returns {Promise<API.Result<API.Unit, API.WriteError>>}
+   */
+  async write({ data, index, source }) {
+    const message = { data, source, index: [...index] }
+    const save = Store.write(this.connection.store, message)
```

Comment:
There is some non compatibility with previous solution that is making `/receipt` endpoint not working anymore. [Here is the lambda for receipt endpoint](https://github.com/w3s-project/w3infra/blob/main/upload-api/functions/receipt.js). In short, it reads from invocation bucket (baf....in) and redirects to the workflow bucket. So, with the new key paths it actually works, but what is being stored in the workflow looks somewhat different now compared to before.

This is how we previously were storing the Body of the Message request: 
- https://github.com/w3s-project/w3infra/blob/main/upload-api/ucan-invocation.js#L71-L76
- https://github.com/w3s-project/w3infra/blob/main/upload-api/ucan-invocation.js#L73
- https://github.com/w3s-project/w3infra/blob/main/upload-api/ucan-invocation.js#L229

while in this PR we are storing the message format above (which actually looks exactly the same).

However, when we try to read workflow from the [filecoin integration tests](https://github.com/w3s-project/w3infra/blob/main/test/filecoin.test.js#L131-L150), now it fails while reading with "'CBOR decode error: encountered invalid minor (31) for major 1'"

Reply:
this was fixed